### PR TITLE
Fix for issue 9229 (GWL)

### PR DESCRIPTION
**Pull request by icarter09**
This PR addresses #9229 

Adjusted the logic to correctly check the values set for the left click action and mouse wheel scroll action.

### CODE REVIEW

**Comment by clefebvre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Complexity increasing per file
==============================
- files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/applet.js  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cinnamon/Cinnamon/pullRequest?prid=6185730&bid=20187250)

File: files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/applet.js [hunk 1]
```diff
@@ -676,78 +676,81 @@ class GroupedWindowListApplet extends Applet.Applet {
...
+            (this.state.settings.leftClickAction !== 3 && this.state.settings.scrollBehavior === 3               
+                        && e && !sourceFromAppGroup)  || 
+            (this.state.settings.leftClickAction === 3 && this.state.settings.scrollBehavior === 3)) {
+
+            this.state.set({scrollactive: true});
+
+            let isappscroll = this.state.settings.scrollbehavior === 2;
+            let direction, source;
+
+            if (sourcefromappgroup) {
```
Comment (mtwebster) [inline files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/applet.js:691]: sourceFromAppGroup?

File: files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/applet.js [hunk 2]
```diff
@@ -676,78 +676,81 @@ class GroupedWindowListApplet extends Applet.Applet {
...
+                isappscroll = false;
+                direction = e ? e.get_scroll_direction() : 1;
+                source = sourcefromappgroup;
```
Comment (mtwebster) [inline files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/applet.js:694]: same

File: files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/applet.js (hunk 1 above)
Comment (icarter09) [reply files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/applet.js:691]: argg vim!  Thanks for the heads up!

File: files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/applet.js (hunk 2 above)
Comment (icarter09) [reply files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/applet.js]: That along with the rest should be squared up now.

**Comment by clefebvre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/applet.js  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cinnamon/Cinnamon/pullRequest?prid=6185730&bid=20187250)

File: files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/applet.js [hunk 3]
```diff
@@ -676,78 +676,81 @@ class GroupedWindowListApplet extends Applet.Applet {
...
-        while ((isAppScroll
-            && (!this.appLists[this.state.currentWs].appList[z]
-                || !this.appLists[this.state.currentWs].appList[z].groupState.lastFocused))
-            || (!isAppScroll &&
-                (!source.groupState.metaWindows[z]
-                    || source.groupState.metaWindows[z] === source.groupState.lastFocused))) {
-            limit--;
-            if (direction === 0) {
-                z -= 1;
+            if (isAppsCroll) {
```
Comment (mtwebster) [inline files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/applet.js:701]: isAppScroll? :)

Comment (icarter09) [reply files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/applet.js]: Fixed

**Comment by clefebvre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- files/usr/share/cinnamon/applets/grouped-window-list@cinnamon.org/applet.js  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/cinnamon/Cinnamon/pullRequest?prid=6185730&bid=20187250)

**Pull request merged**